ts/src/pro/hitbtc.ts

```diff
@@ -14,150 +17,248 @@ export default class hitbtc extends hitbtcRest {
-                    'ohlcv': 'subscribeCandles',
+                'watchTicker': {
+                    'method': 'ticker/{speed}',  // 'ticker/{speed}' or 'ticker/price/{speed}'
                 },
+                'watchTickers': {
+                    'method': 'ticker/{speed}',  // 'ticker/{speed}','ticker/price/{speed}', 'ticker/{speed}/batch', or 'ticker/{speed}/price/batch''
+                },
+                'watchOrderBook': {
+                    'method': 'orderbook/full',  // 'orderbook/full', 'orderbook/{depth}/{speed}', 'orderbook/{depth}/{speed}/batch', 'orderbook/top/{speed}', or 'orderbook/top/{speed}/batch'
+                },
+            },
+            'timeframes': {
+                '1m': 'M1',
+                '3m': 'M3',
+                '5m': 'M5',
+                '15m': 'M15',
+                '30m': 'M30',
+                '1h': 'H1',
+                '4h': 'H4',
+                '1d': 'D1',
+                '1w': 'D7',
+                '1M': '1M',
+            },
+            'streaming': {
+                'keepAlive': 4000,
             },
         });
     }
 
-    async watchPublic (symbol: string, channel, timeframe = undefined, params = {}) {
+    async authenticate () {
+        /**
+         * @ignore
+         * @method
+         * @description authenticates the user to access private web socket channels
+         * @see https://api.hitbtc.com/#socket-authentication
+         * @returns {object} response from exchange
+         */
+        this.checkRequiredCredentials ();
+        const url = this.urls['api']['ws']['private'];
+        const messageHash = 'authenticated';
+        const client = this.client (url);
+        let future = this.safeValue (client.subscriptions, messageHash);
+        if (future === undefined) {
+            const timestamp = this.milliseconds ();
+            const signature = this.hmac (this.encode (this.numberToString (timestamp)), this.encode (this.secret), sha256, 'hex');
+            const request = {
+                'method': 'login',
+                'params': {
+                    'type': 'HS256',
+                    'api_key': this.apiKey,
+                    'timestamp': timestamp,
+                    'signature': signature,
+                },
+            };
+            future = await this.watch (url, messageHash, request);
+            //
+            //    {
+            //        jsonrpc: '2.0',
+            //        result: true
+            //    }
+            //
+            //    # Failure to return results
+            //
+            //    {
+            //        jsonrpc: '2.0',
+            //        error: {
+            //            code: 1002,
+            //            message: 'Authorization is required or has been failed',
+            //            description: 'invalid signature format'
+            //        }
+            //    }
+            //
+            client.subscriptions[messageHash] = future;
```

Comment:
@samgermain let's avoid storing futures inside `client.subscriptions` because they're not iterable and that might be a problem in the future.

Can you follow this new approach instead? https://github.com/ccxt/ccxt/pull/18917